### PR TITLE
initial version manager fuzz test and testing doc

### PR DESCRIPTION
**Pull request by eggyhead**
### What are we doing? 
Adding fuzz testing to the `rotational` versions test suite to increase coverage of edge cases.

### How are we doing it?
- [x] Adding `go-fuzz-headers` library to help with using structs in our test 
- [x] Adding tests for the versions methods _note: we're only testing that any of the configs created by the fuzzer do not throw an error, we're not asserting any specific output here_
- [x] Adding docs on how to run the tests 

### How do I test? 
follow the instructions in the test doc :smile:

### CODE REVIEW

File: docs/testing.md [hunk 1]
```diff
@@ -23,4 +23,4 @@ Since fuzzing involves generating inputs at time intervals, you explicitly run t
 Watch the output, and view any errors that may arise. When you are satisfied the tests confirm your code working as expected, you can exit the test. 
 
 eg. 
-` go test -fuzz=FuzzNewVersionManager -v`
+` go test -fuzz=FuzzVersionManager_Delete -v`
```
Comment (bbengfort) [inline docs/testing.md:26]: Do you think it would be a good idea to update GitHub actions with fuzzing? Alternatively, we could create a GitHub action that PRs don't depend on that performs fuzzing.

File: go.mod [hunk 2]
```diff
@@ -3,11 +3,12 @@ module github.com/rotationalio/honu
 go 1.16
```
Comment (bbengfort) [inline go.mod:3]: We'll have to update this to go 1.19 for the tests to pass!

**Review by bbengfort — COMMENTED** [review]
Thank you so much @eggyhead for continuing to push on this! I wasn't sure if you were ready for a review or not, so just wanted to check in. If you are ready for a review, I'll do a more thorough reading of the code, but everything looks great so far!

File: docs/testing.md (hunk 1 above)
Comment (eggyhead) [reply docs/testing.md:26]: Hi! Is this question about whether these tests will run automatically? 

Currently, the fuzz tests will need to be run manually, as described in the doc.

Comment (bbengfort) [reply docs/testing.md:26]: Makes sense, thank you!

**Comment by bbengfort** [comment]
@eggyhead I'm sorry I haven't responded in a while, I wasn't sure if this was ready for review or not; things are looking great though and we're really excited to merge this PR - let me know!
